examples/scripts/04-ppo-sentiment-accelerate.py

```diff
@@ -0,0 +1,179 @@
+
+# We first define the configuration of the experiment, defining the model, the dataset,
+# the training parameters, and the PPO parameters.
+config = {
+    "model_name": "lvwerra/gpt2-imdb",
+    "dataset_name": "imdb",
+    "cls_model_name": "lvwerra/distilbert-imdb",
+    "steps": 20000,
+    "batch_size": 256,
+    "forward_batch_size": 16,
+    "ppo_epochs": 4,   
+    "txt_in_min_len": 2,
+    "txt_in_max_len": 8,
+    "txt_out_min_len": 4,
+    "txt_out_max_len": 16,
+    "lr": 1.41e-5,
+    "init_kl_coef":0.2,
+    "target": 6,
+    "horizon":10000,
+    "gamma":1,
+    "lam":0.95,
+    "cliprange": .2,
+    "cliprange_value":.2,
+    "vf_coef":.1, 
+}
+
+# We then define the arguments to pass to the sentiment analysis pipeline.
+# We set `return_all_scores` to True to get the sentiment score for each token.
+sent_kwargs = {
+    "return_all_scores": True,
+    "function_to_apply": "none",
+    "batch_size": config["forward_batch_size"]
+}
+
+# Below is an example function to build the dataset. In our case, we use the IMDB dataset
+# from the `datasets` library. One should customize this function to train the model on
+# its own dataset.
+def build_dataset(config):
+    """
+    Build dataset for training. This builds the dataset from `load_dataset`, one should 
+    customize this function to train the model on its own dataset.
+    
+    Args:
+        dataset_name (`str`): 
+            The name of the dataset to be loaded.
+    
+    Returns:
+        dataloader (`torch.utils.data.DataLoader`):
+            The dataloader for the dataset.
+    """
+    dataset_name = config["dataset_name"]
+    tokenizer = AutoTokenizer.from_pretrained(config["model_name"])
+    tokenizer.pad_token = tokenizer.eos_token
+    # load imdb with datasets
+    ds = load_dataset(dataset_name, split='train')
+    ds = ds.rename_columns({'text': 'review', 'label': 'sentiment'})
+    ds = ds.filter(lambda x: len(x["review"])>200, batched=False)
+
+    input_size = LengthSampler(config["txt_in_min_len"], config["txt_in_max_len"])
+
+    def tokenize(sample):
+        sample["tokens"] = tokenizer.encode(sample["review"])[:input_size()]
+        sample["query"] = tokenizer.decode(sample["tokens"])
+        return sample
+
+    ds = ds.map(tokenize, batched=False)
+
+    def collater(data):
+        return dict((key, [d[key] for d in data]) for key in data[0])
+
+    dataloader = torch.utils.data.DataLoader(ds, batch_size=config['batch_size'], collate_fn=collater)
+    return dataloader
+
+# We retrieve the dataloader by calling the `build_dataset` function.
+dataloader = build_dataset(config)
+
+# Now let's build the model, the reference model, and the tokenizer.
+model = AutoModelForCausalLMWithValueHead.from_pretrained(config["model_name"])
+ref_model = AutoModelForCausalLMWithValueHead.from_pretrained(config["model_name"])
+tokenizer = AutoTokenizer.from_pretrained(config["model_name"])
+
+tokenizer.pad_token = tokenizer.eos_token
```

Comment:
what is the reason behind this?

Reply:
Fixed in [747d5f0](https://github.com/lvwerra/trl/pull/58/commits/747d5f021f5dcf16735930916590bcaac6d1a5fb) appears to be an issue only for gpt2